lxd/resource_lxd_network_zone.go

```diff
@@ -0,0 +1,241 @@
+package lxd
+
+import (
+	"log"
+
+	"github.com/canonical/lxd/shared/api"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceLxdNetworkZone() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLxdNetworkZoneCreate,
+		Update: resourceLxdNetworkZoneUpdate,
+		Delete: resourceLxdNetworkZoneDelete,
+		Exists: resourceLxdNetworkZoneExists,
+		Read:   resourceLxdNetworkZoneRead,
+		Importer: &schema.ResourceImporter{
+			State: resourceLxdNetworkZoneImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"remote": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "",
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"config": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceLxdNetworkZoneCreate(d *schema.ResourceData, meta interface{}) error {
+	p := meta.(*lxdProvider)
+	server, err := p.GetInstanceServer(p.selectRemote(d))
+	if err != nil {
+		return err
+	}
+
```

Comment:
Can you please just add a check if server supports `projects_networks_zones` extension:

```go
netZoneExt := "projects_networks_zones"
if !server.HasExtension(netZoneExt) {
	return fmt.Errorf("Network zones cannot be used because LXD server does not support %q extension", netZoneExt)
}
```